logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt

```diff
@@ -194,4 +193,31 @@ class DefaultPathAssertions : PathAssertions {
                     container.resolve(entry).collect { _logicAppend { exists(NOFOLLOW_LINKS) } }
                 }
         ).build()
+
+    override fun <T : Path> isEmptyDirectory(container: AssertionContainer<T>): Assertion =
+        if (container.isDirectory().holds()) {
+            container.changeSubject.unreported {
+                it.runCatchingIo { Files.list(it).findAny() }
```

Comment:
As written in the issue description, you should use `.use{...}` in order that the stream is closed. 

IMO the current implementation is fine as first step. If you like, then improve it by showing more than just one file. maybe 7 and if there are more you use something like `...`. If you want to do this, then you need to switch from assertionBuilder.descriptive to assertionBuilder.list and then you can use assertionBuilder.representationOnly inside to list the different files.

Reply:
The implementation now uses .use{...}